Assets/Scripts/UI/UIP2PTransportMenu.cs

```diff
@@ -0,0 +1,299 @@
+{
+    public class P2PTransportPresenceData
+    {
+        public const string ValidIdenfier = "P2PTRANSPORT";
+
+        public string SceneIdentifier;
+        public string ServerUserId;
+
+        public bool IsValid()
+        {
+            return SceneIdentifier == ValidIdenfier;
+        }
+    }
+
+    public class UIP2PTransportMenu : UIFriendInteractionSource, ISampleSceneUI
+    {
+        public UIFriendsMenu FriendUI;
+        public GameObject PlayerNetworkPrefab;
+        public Image Background;
+        public GameObject StartHostButton;
+        public GameObject DisconnectButton;
+        public GameObject MovementControls;
+        public RectTransform DisplayNameContainer;
+        public float MovementStepAmount = 0.5f;
+
+        private EOSTransportManager transportManager = null;
+        private bool isHost = false;
+        private bool isClient = false;
+        private bool uiDirty = false;
+
+        private ulong joinGameAcceptedNotifyHandle = 0;
+
+        private readonly Vector2[] movementVectors = new Vector2[] { Vector2.up, Vector2.left, Vector2.down, Vector2.right };
+
+        private void Start()
+        {
+            transportManager = EOSManager.Instance.GetOrCreateManager<EOSTransportManager>();
+        }
+
+        public override FriendInteractionState GetFriendInteractionState(FriendData friendData)
+        {
+            if (isHost || isClient || string.IsNullOrEmpty(friendData.Presence?.JoinInfo))
+            {
+                return FriendInteractionState.Hidden;
+            }
+
+            var joinInfo = JsonUtility.FromJson<P2PTransportPresenceData>(friendData.Presence.JoinInfo);
+            return joinInfo.IsValid() ? FriendInteractionState.Enabled : FriendInteractionState.Hidden;
+        }
+
+        public override string GetFriendInteractButtonText()
+        {
+            return "Join";
+        }
+
+        public override void OnFriendInteractButtonClicked(FriendData friendData)
+        {
+            var joinInfo = JsonUtility.FromJson<P2PTransportPresenceData>(friendData.Presence.JoinInfo);
+            if (joinInfo.IsValid())
+            {
+                var hostId = ProductUserId.FromString(joinInfo.ServerUserId);
+                if (hostId.IsValid())
+                {
+                    JoinGame(hostId);
+                }
+                else
+                {
+                    Debug.LogError("UIP2PTransportMenu (OnFriendInteractButtonClicked): invalid server user id");
+                }
+            }
+            else
+            {
+                Debug.LogError("UIP2PTransportMenu (OnFriendInteractButtonClicked): invalid join info");
+            }
+        }
+
+        public override bool IsDirty()
+        {
+            return uiDirty;
+        }
+
+        public override void ResetDirtyFlag()
+        {
+            uiDirty = false;
+        }
+
+        public void StartHostOnClick()
+        {
+            if (isHost)
```

Comment:
I'm assuming that it's not possible to have `isHost` and `isClient` both true due to how the UI is connected since there's no checks for it.

Reply:
Yes, being a host and client is mutually exclusive and becoming either disables all UI for becoming or joining a host.

Reply:
I couldn't find any cases where multiple friend actions were available simultaneously. Plus, all the old buttons were stacked on top of each other so it wouldn't have working anyways. If that is wanted later we can add support then.